tools/image-audit/promisepool.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * A class that manages a pool of promises with a maximum concurrency limit.
+ * It controls how many promises are running at any time, and queues any additional promises.
+ */
+/* eslint-disable no-console */
+class PromisePool {
```

Comment:
I understand you want to manage promises concurrency but this looks overly complicated :-)